src/benchmarklib/benchmark_runner.cpp

```diff
@@ -323,8 +336,24 @@ void BenchmarkRunner::_execute_query(const QueryID query_id, const std::function
   auto pipeline_builder = SQLPipelineBuilder{sql}.with_mvcc(_config.use_mvcc);
   if (_config.enable_visualization) pipeline_builder.dont_cleanup_temporaries();
   auto pipeline = pipeline_builder.create_pipeline();
-  // Execute the query, we don't care about the results
-  pipeline.get_result_table();
+
+  if (!_config.validate) {
+    // Execute the query, we don't care about the results
+    pipeline.get_result_table();
+  } else {
+    const auto hyrise_result = pipeline.get_result_table();
+    const auto sqlite_result = _sqlite_wrapper->execute_query(sql);
+
+    // check_table_equal does not handle empty tables well    
```

Comment:
Can we move that check for empty tables into check_table_equal? I think it would be in better hands there.